app.py

```diff
@@ -45,24 +67,44 @@ def get_task(task_id):
         return jsonify({'message': 'Task not found'}), 404
 
 # Route to update the task
-@app.route('/update/tasks/<task_id>', methods=['PUT'])
+# Route for update task
+@app.route('/tasks/<task_id>', methods=['PUT'])
 def update_task(task_id):
-    data = request.get_json()
-    update_data = {k: task_schema[k](v) for k, v in data.items() if k in task_schema}
-    mongo.db.tasks.update_one({'_id': ObjectId(task_id)}, {'$set': update_data})
-    return jsonify({'message': 'Task updated successfully'})
+    try:
+        data = request.get_json()
+        if 'name' in data and not data['name']:
```

Comment:
Both of the conditions do the same thing.
If before end is True, Then after and will never be False.